server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

```diff
@@ -1190,7 +1187,13 @@ public void testResolveLifecycle() throws Exception {
         // Composable Z: -
         // Result: "lifecycle": {"enabled": true}, here the result of the composition is with retention explicitly
         // nullified, but effectively this is equivalent to infinite retention.
-        assertLifecycleResolution(service, project, List.of(ct30d, ctNullRetention), null, DataStreamLifecycle.Template.DATA_DEFAULT);
+        assertLifecycleResolution(
+            service,
+            project,
+            List.of(ct30d, ctNullRetention),
+            null,
+            DataStreamLifecycle.dataLifecycleBuilder().dataRetention(ResettableValue.reset()).buildTemplate()
+        );
```

Comment:
The assertions in this test were previously technically incorrect. With my change to `ResettableValue` in the builder, we can now properly assess these values. Previously, we weren't able to distinguish between `ResettableValue.reset()` and `ResettableValue.undefined()` because of how the template composition uses the builders (which wasn't necessarily a problem, as that doesn't matter for the final value, which is `null` in both cases), but now the end result does represent the correct intention of the value.